src/v/cloud_storage_clients/client_pool.cc

```diff
@@ -426,6 +426,33 @@ client_pool::acquire(ss::abort_source& as) {
     co_return lease;
 }
 
+auto client_pool::acquire_with_timeout(
+  ss::abort_source& as,
+  ss::lowres_clock::time_point deadline,
+  std::optional<ss::sstring> ctx) -> ss::future<client_lease> {
+    auto lease = co_await acquire(as);
+    // take a copy of the shared_ptr held by the lease to avoid racing with
+    // client_pool teardown
+    auto to = deadline - ss::lowres_clock::now();
```

Comment:
nit: would it be helpful to expose `watchdog` constructor accepting `time_point` ?

Reply:
maybe so. for the logging, i think the `duration` is more helpful than the `time_point`, so it feels like the same same.